packages/atomic/src/components/common/item-link/item-link.ts

```diff
@@ -0,0 +1,133 @@
+import {FunctionalComponentWithChildren} from '@/src/utils/functional-component-utils';
+import {html} from 'lit';
+import {ifDefined} from 'lit/directives/if-defined.js';
+import {ref} from 'lit/directives/ref.js';
+import {filterProtocol} from '../../../utils/xss-utils.js';
+import {ItemTarget} from '../layout/display-options.js';
+
+interface ItemLinkEventProps {
+  onSelect: () => void;
+  onBeginDelayedSelect: () => void;
+  onCancelPendingSelect: () => void;
+  stopPropagation?: boolean;
+}
+
+export interface ItemLinkProps extends ItemLinkEventProps {
+  href: string;
+  className?: string;
+  part?: string;
+  title?: string;
+  ref?: (elm?: HTMLAnchorElement) => void;
+  stopPropagation?: boolean;
+  attributes?: Attr[];
+  tabIndex?: number;
+  ariaHidden?: boolean;
+  target?: ItemTarget;
+  rel?: string;
+  onMouseOver?: () => void;
+  onMouseLeave?: () => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+}
+
+export const bindAnalyticsToLink = (
+  link: HTMLAnchorElement,
+  {
+    onSelect,
+    onBeginDelayedSelect,
+    onCancelPendingSelect,
+    stopPropagation = true,
+  }: ItemLinkEventProps
+) => {
+  const stopPropagationAndProcess = (e: Event, process: () => void) => {
+    stopPropagation && e.stopPropagation();
+    process();
+  };
+  (['click', 'contextmenu', 'mousedown', 'mouseup'] as const).forEach(
+    (eventName) =>
+      link.addEventListener(eventName, (e) =>
+        stopPropagationAndProcess(e, onSelect)
+      )
+  );
+  link.addEventListener(
+    'touchstart',
+    (e) => stopPropagationAndProcess(e, onBeginDelayedSelect),
+    {passive: true}
+  );
+  link.addEventListener(
+    'touchend',
+    (e) => stopPropagationAndProcess(e, onCancelPendingSelect),
+    {passive: true}
+  );
+};
+
+export const renderLinkWithItemAnalytics: FunctionalComponentWithChildren<
+  ItemLinkProps
+> =
+  ({props}) =>
+  (children) => {
+    const {
+      href,
+      className,
+      part,
+      title,
+      onSelect,
+      onBeginDelayedSelect,
+      onCancelPendingSelect,
+      onMouseOver,
+      onMouseLeave,
+      onFocus,
+      onBlur,
+      ref: refCallback,
+      attributes,
+      tabIndex,
+      ariaHidden,
+      rel,
+      stopPropagation = true,
+      target = '_self',
+    } = props;
+
+    return html`
+      <a
+        class=${className ?? ''}
```

Comment:
For the props where we default to an empty string, we could instead use the `ifDefined` Lit directive.